include/onnxruntime/core/graph/node_arg.h

```diff
@@ -67,15 +67,17 @@ class NodeArg {
   /** Validate and merge type [and shape] info from input_type.
   @param strict If true, the shape update will fail if there are incompatible values. 
                 If false, will be lenient and merge only shape info that can be validly processed.
+  @param override_types If true, resolve the two inputs or two outputs type when different
```

Comment:
how about `override_type`? it's only changing one type here right?
and maybe something like `If true, override type with input type.`

Reply:
hmm, but it's for many types in one option. debatable, will leave it.